reana_workflow_controller/rest.py

```diff
@@ -1509,7 +1509,9 @@ def _get_workflow_with_uuid_or_name(uuid_or_name):
         # Search by `run_number` since it is a primary key.
         # workflow = Workflow.query. \
         #     filter(Workflow.name == run_number).first()
-        workflow = Workflow.query.get(run_number)
+        workflow = Workflow.query.filter(Workflow.name == workflow_name,
+                                         Workflow.run_number == run_number).\
```

Comment:
We should probably explicitly include a test for `owner_id` in case several people use the same workflow name... especially in case we delete the auto-incremented run number...